### PR TITLE
Better error messages

### PR DESCRIPTION
**Pull request by geek1011**

### CODE REVIEW

File: kobo-uncaged/main.go [hunk 1]
```diff
@@ -591,7 +591,7 @@ func (ku *KoboUncaged) updateNickelDB() error {
 		}
 		_, err = stmt.Exec(desc, series, seriesNum, seriesNumFloat, cid)
 		if err != nil {
-			log.Print(err)
+			log.Print(wrapPos(wrap(err, "could exec nickel db update with %#v", []interface{}{desc, series, seriesNum, seriesNumFloat, cid})))
```
Comment (NiLuJe) [inline kobo-uncaged/main.go]: `could` -> `could not`?

Comment (geek1011) [reply kobo-uncaged/main.go]: Oops ... I've done that so many times before :smile:

File: kobo-uncaged/util.go [hunk 2]
```diff
@@ -74,3 +76,21 @@ func readJSON(fn string, out interface{}) (emptyOrNotExist bool, err error) {
 
 	return false, json.NewDecoder(f).Decode(out)
 }
+
+func wrapPos(err error) error {
+	if err != nil {
+		pc, file, line, ok := runtime.Caller(0)
```
Comment (geek1011) [inline kobo-uncaged/util.go]: Oops

File: kobo-uncaged/util.go [hunk 3]
```diff
@@ -74,3 +76,21 @@ func readJSON(fn string, out interface{}) (emptyOrNotExist bool, err error) {
...
+		pc, file, line, ok := runtime.Caller(1)
+		if !ok {
+			return wrap(err, "[unknown pos]")
+		}
+		return wrap(err, "[0x%X %s:%d]", pc, file, line)
+	}
+	return nil
+}
+
+func wrap(err error, format string, a ...interface{}) error {
```
Comment (kobothingy) [inline kobo-uncaged/util.go]: Why wrap an error instead of fixing it?
```suggestion
func fix(err error) error {
    return nil
}
```

**Comment by geek1011** [comment]
I'm closing this for now, as it's been a while. I'll work on this again later. Also, there has been and will be more changes in KU before I have time to work on this, so it'll be easier to start again.

**Pull request closed without merging**